pkg/controllers/subnet/subnet_controller.go

```diff
@@ -76,7 +78,23 @@ func (r *SubnetReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 				obj.Spec.IPv4SubnetSize = vpcNetworkConfig.DefaultIPv4SubnetSize
 			}
 		}
-		if _, err := r.Service.CreateOrUpdateSubnet(obj, nil); err != nil {
+
+		var tags []model.Tag
+		namespace := &v1.Namespace{}
+		namespacedName := types.NamespacedName{
+			Name: req.Namespace,
+		}
+		if err := r.Client.Get(context.Background(), namespacedName, namespace); err != nil {
+			log.Error(err, "unable to fetch namespace of Subnet CR", "req", req.NamespacedName)
+			updateFail(r, &ctx, obj)
+			return ResultRequeue, err
+		}
+		namespace_uid := namespace.UID
+		tags = append(tags,
+			model.Tag{Scope: servicecommon.String(servicecommon.TagScopeNamespaceUID), Tag: servicecommon.String(string(namespace_uid))},
```

Comment:
why we add  TagScopeNamespaceUID and TagScopeNamespace tags in reconcile logic? It looks like it will be added for all subnets, including VM subnet and Pod subnet? 

Also, from https://github.com/vmware-tanzu/nsx-operator/pull/304/, TagScopeNamespaceUID and TagScopeNamespace will become the basic tags for all subnet in the builder basicTags, if that is true, there is no need to add here.

Reply:
Here only creates VM subnet,  the default Pod subnet is created by subnetport.
OK, removed.

Reply:
Since https://github.com/vmware-tanzu/nsx-operator/pull/304 is merged, in #304, 
https://github.com/vmware-tanzu/nsx-operator/blob/vpc_dev/pkg/util/utils.go#L353
TagScopeNamespaceUID and TagScopeNamespace will become the basic tags, you can submit a new patch later to remove TagScopeNamespaceUID and TagScopeNamespace  in the util.BuildBasicTags.

In addition, why we only add TagScopeNamespaceUID here? we also need TagScopeNamespace.

I see you also add TagScopeNamespaceUID at https://github.com/vmware-tanzu/nsx-operator/pull/298/files#diff-b596db2096eee6bc223e30c78f604cf222521d3313ba43176e82bb0ce710d16aR95?  why add TagScopeNamespaceUID in two places?

Reply:
OK, would submit a new patch to refactor tags for subnet/subnetset.